meson.build

```diff
@@ -0,0 +1,179 @@
+        INSTALL_DIR / 'crossover'
+    ],
+    [
+        'broken_pairs_distance',
+        [
+            SRC_DIR / 'diversity' / 'broken_pairs_distance.cpp',
+            SRC_DIR / 'Individual.cpp',
+            SRC_DIR / 'ProblemData.cpp',
+            SRC_DIR / 'PenaltyManager.cpp',
+            SRC_DIR / 'XorShift128.cpp',
+        ],
+        INSTALL_DIR / 'diversity'
+    ],
+    [
+        'LocalSearch',
+        [
+            SRC_DIR / 'educate' / 'LocalSearch.cpp',
+            SRC_DIR / 'educate' / 'Route.cpp',
+            SRC_DIR / 'educate' / 'Node.cpp',
+            SRC_DIR / 'ProblemData.cpp',
+            SRC_DIR / 'Individual.cpp',
+            SRC_DIR / 'PenaltyManager.cpp',
+            SRC_DIR / 'XorShift128.cpp',
+        ],
+        INSTALL_DIR / 'educate'
+    ],
+    [
+        'Exchange',
+        [
+            SRC_DIR / 'educate' / 'Exchange.cpp',
+            SRC_DIR / 'educate' / 'Node.cpp',
+            SRC_DIR / 'ProblemData.cpp',
+            SRC_DIR / 'PenaltyManager.cpp',
+        ],
+        INSTALL_DIR / 'educate'
+    ],
+    [
+        'MoveTwoClientsReversed',
+        [
+            SRC_DIR / 'educate' / 'MoveTwoClientsReversed.cpp',
+            SRC_DIR / 'educate' / 'Node.cpp',
+            SRC_DIR / 'ProblemData.cpp',
+            SRC_DIR / 'PenaltyManager.cpp',
+        ],
+        INSTALL_DIR / 'educate'
+    ],
+    [
+        'TwoOpt',
+        [
+            SRC_DIR / 'educate' / 'TwoOpt.cpp',
+            SRC_DIR / 'educate' / 'Node.cpp',
+            SRC_DIR / 'ProblemData.cpp',
+            SRC_DIR / 'PenaltyManager.cpp',
+        ],
+        INSTALL_DIR / 'educate'
+    ],
+    [
+        'RelocateStar',
+        [
+            SRC_DIR / 'educate' / 'RelocateStar.cpp',
+            SRC_DIR / 'educate' / 'Exchange.cpp',
+            SRC_DIR / 'educate' / 'Route.cpp',
+            SRC_DIR / 'educate' / 'Node.cpp',
+            SRC_DIR / 'ProblemData.cpp',
+            SRC_DIR / 'PenaltyManager.cpp',
+        ],
+        INSTALL_DIR / 'educate'
+    ],
+    [
+        'SwapStar',
+        [
+            SRC_DIR / 'educate' / 'SwapStar.cpp',
+            SRC_DIR / 'educate' / 'Route.cpp',
+            SRC_DIR / 'educate' / 'Node.cpp',
+            SRC_DIR / 'ProblemData.cpp',
+            SRC_DIR / 'PenaltyManager.cpp',
+        ],
+        INSTALL_DIR / 'educate'
+    ]
+]
+
+foreach extension : extensions
+    message('Going to build ' + extension[0])
+
+    py.extension_module(
+        extension[0],
+        extension[1],
+        dependencies : [py.dependency(), pybind11],
+        install : true,
+        install_dir: extension[2],
+        include_directories: [
+            include_directories(SRC_DIR),
+        ],
+        cpp_args: [
+            '-Werror',
+            '-Wall',
+            '-Wextra',
+            '-Wpedantic',
+            # Ignore unused parameters, because not all LS operators use every
+            # argument they are given.
+            '-Wno-unused-parameter'
```

Comment:
I would set -Werror as `default_options: ['werror=true']`.

-Wall is already the default because the default `warning_level=1` option sets that. But you can add `default_options: ['warning_level=3']`, 2 sets -Wextra and 3 sets -Wpedantic.

The only one you need to set manually is -Wno-unused-parameter. This won't work on Windows compilers, but you could use [`compiler.get_supported_arguments()`](https://mesonbuild.com/Reference-manual_returned_compiler.html#compilerget_supported_arguments) to check if it's supported.

(Using the builtin core options for werror/warning_level gets you cross-platform flag handling for free.)